builtins/target-avx512skx-x32.ll

```diff
@@ -56,7 +56,100 @@ define i1 @__none(<WIDTH x MASK> %mask) nounwind readnone alwaysinline {
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;; broadcast/rotate/shift/shuffle
 
-define_shuffles()
+; Use vpermw for 16-bit shuffles.
+declare <WIDTH x i16> @llvm.x86.avx512.mask.permvar.hi.512(<WIDTH x i16>, <WIDTH x i16>, <WIDTH x i16>, i16)
+define <WIDTH x i16> @__shuffle_i16(<WIDTH x i16>, <WIDTH x i32>) nounwind readnone alwaysinline {
+  %ind = trunc <WIDTH x i32> %1 to <WIDTH x i16>
+  %res = call <WIDTH x i16> @llvm.x86.avx512.mask.permvar.hi.512(<WIDTH x i16> %0, <WIDTH x i16> %ind, <WIDTH x i16> zeroinitializer, i16 -1)
+  ret <WIDTH x i16> %res
+}
+
+define <WIDTH x i8> @__shuffle_i8(<WIDTH x i8>, <WIDTH x i32>) nounwind readnone alwaysinline {
+  %vals = zext <WIDTH x i8> %0 to <WIDTH x i16>
```

Comment:
I don't think we can safely use zext/trunc here since they don't preserve the exact bit patterns. Same for x64 target.

Reply:
Right, I was mistaken, the original bit pattern is indeed preserved